### PR TITLE
more clear error prompt of mismatched crypto type

### PR DESCRIPTION
**Pull request by sulenn**
determine whether the FISCO-BCOS Version of Node is consistent with SMCrypto configuration item when init client

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/go_client.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5903866&bid=19333682)

File: client/go_client.go [hunk 1]
```diff
@@ -82,23 +82,38 @@ func DialContext(ctx context.Context, config *conf.Config) (*Client, error) {
...
 		return nil, fmt.Errorf("DialContext failed, err: %v", err)
 	}
+
+	// determine whether FISCO-BCOS Version is consistent with SMCrypto configuration item
+	var fiscoBcosVersion string
+	fiscoBcosVersion, ok = m["FISCO-BCOS Version"].(string)
+	if !ok {
+		return nil, errors.New("JSON response does not contains the key : FISCO-BCOS Version")
+	}
+	if strings.Contains(fiscoBcosVersion, "gm") != config.IsSMCrypto {
```
Comment (bxq2011hust) [inline client/go_client.go:103]: can ignore the “gm” case？GM/gm all are valid

File: client/go_client.go [hunk 2]
```diff
@@ -82,23 +82,38 @@ func DialContext(ctx context.Context, config *conf.Config) (*Client, error) {
...
+		return nil, errors.New("the connected node does not support the encryption method in the configuration item")
```
Comment (bxq2011hust) [inline client/go_client.go:104]: the SDK set SMCrypt=%v, but the node is mismatched, config.IsSMCrypto

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/go_client.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5903866&bid=19333682)

**Pull request merged**